pkg/etcdenvvar/etcd_env.go

```diff
@@ -255,6 +269,35 @@ func getCipherSuites(envVarContext envVarContext) (map[string]string, error) {
 	}, nil
 }
 
+// getMaxLearners reads the control-plane replicas from the install-config. We tolerate max learners equal to the
+// desired control-plane replicas so that the admin can surge up N x 2 in case of vertical scaling/replacement.
+func getMaxLearners(envVarContext envVarContext) (map[string]string, error) {
+	clusterConfig, err := envVarContext.configmapLister.ConfigMaps(operatorclient.TargetNamespace).Get(clusterConfigName)
+	if err != nil {
+		errMsg := fmt.Errorf("failed to get configmap %s/%s :%v", operatorclient.TargetNamespace, clusterConfigName, err)
+		klog.Error(errMsg)
+		return nil, errMsg
+	}
+
+	d := replicaCountDecoder{}
+	if err := yaml.Unmarshal([]byte(clusterConfig.Data[clusterConfigKey]), &d); err != nil {
+		errMsg := fmt.Errorf("%s key doesn't exist in configmap %s/%s :%w", clusterConfigName, operatorclient.TargetNamespace, clusterConfigKey, err)
```

Comment:
2nd time's the charm 😄 
```suggestion
		errMsg := fmt.Errorf("%s key doesn't exist in configmap %s/%s :%w", clusterConfigKey, operatorclient.TargetNamespace, clusterConfigName, err)
```

Reply:
:man_facepalming:

Reply:
updated....